File: Binary Tree/102. Binary Tree Level Order Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution
{
public:
    vector<vector<int>> levelOrder(TreeNode *root)
    {

        vector<vector<int>> traversal;
        if (root == NULL)
        {
            return traversal;
        }

        queue<TreeNode *> q;

        q.push(root);

        while (!q.empty())
        {

            int qlen = q.size();
            vector<int> row;

            for (int i = 0; i < qlen; i++)
            {

                TreeNode *node = q.front();
                row.push_back(node->val);
                q.pop();

                if (node->left != NULL)
                    q.push(node->left);

                if (node->right != NULL)
                    q.push(node->right);
            }

            traversal.push_back(row);
        }

        return traversal;
    }
};

File: Binary Tree/102. Binary Tree Level Order Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> ans=new ArrayList<List<Integer>>();
        Queue <TreeNode> q=new LinkedList<TreeNode>();
        if(root==null)
            return ans;
        q.offer(root);
        q.offer(null);
        List<Integer> arr=new ArrayList<Integer>();
        while(q.size()!=0)
        {
            TreeNode curr=q.poll();
            if(curr==null)
            {
                if(q.size()!=0)
                {
                    ans.add(arr);
                    arr=new ArrayList<Integer>();
                    q.offer(null);
                }
            }
            else
            {
                if(curr.left!=null)
                    q.offer(curr.left);
                if(curr.right!=null)
                    q.offer(curr.right);
                arr.add(curr.val);
                //System.out.println(curr.val);
            }
        }
        ans.add(arr);
        return ans;
            
    }
}